src/instructlab/lab.py

```diff
@@ -1041,19 +1043,31 @@ def train(
             )
             raise click.exceptions.Exit(1)
 
-    if not utils.is_macos_with_m_chip():
+    if backend == "pytorch":
         # Local
         from .llamacpp.llamacpp_convert_to_gguf import convert_llama_to_gguf
-        from .train.linux_train import linux_train
+        from .train.pytorch_train import pytorch_train
 
-        linux_train(
-            ctx=ctx,
+        if device.type == "mps":
+            if (
+                "PYTORCH_MPS_HIGH_WATERMARK_RATIO" not in os.environ
+                or "PYTORCH_ENABLE_MPS_FALLBACK" not in os.environ
```

Comment:
Why are these needed as env variables? Why do users have to explicitly set them? Why can't we handle that on our own?